security/fma-withdrawals-root-in-block-header.md

```diff
@@ -0,0 +1,134 @@
+- [monorepo PR](https://github.com/ethereum-optimism/optimism/pull/13962)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Withdrawals downtime do to inaccurate `withdrawalsRoot`
+
+- **Description:**
+  If the `withdrawalsRoot` in the block header is incorrect, critical infra used to enable withdrawals may fail. Namely, output proposals and challenges would be incorrect, affecting chains with permissioned and chains with permissionless proofs. This is because these components will, with the activation of the Isthmus fork, use the `withdrawalsRoot` header field instead of querying the information from an archive node in the usual way. Output roots are returned by the op-node [`optimism_outputAtBlock`](https://docs.optimism.io/operators/node-operators/json-rpc#optimism_outputatblock) RPC method, and this behaves differently under Isthmus -- when handling a request for an output root (it no longer delegates an `eth_getProof` to op-geth and just reads the information from the block header).
+
+  Triggers:
+
+  - A failed hardfork activation in the execution client.
+
+  - If there is an execution client bug, for example it is possible the root is (incorrectly) added to the header before the state is fully committed.
+
+  - If we were to ever introduce non empty `withdrawals` in the block body, this might override the mechanism introduced with this feature and invalidate the interpreation of the `withdrawalsRoot` field.
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  Temporary downtime for withdrawals, or loss of funds if not remediated in time to challenge a malicious proposal.
+
+  **Mitigations:**
+
+  - We rely on e2e tests to check for consistency between the outputs returned from op-node and those constructed manually in the old way.
+
+  - Instead of waiting for the failure mode to materialize and then writing a patch in a rush, we could add an optional config var to op-node to switch it back into the old behavior. Rolling out the fix would not then require any software releases.
+
+  - op-geth could be modified to log a critical error (triggering an alert) if the withdrawals list in the body is ever non empty.
+
+- **Detection:**
+  Fault proof monitoring systems may not detect this failure mode immediately, until an actor running patched software made a proposal or challenge.
+
+- **Recovery Path(s)**:
+  Fault proof infra would nee to be pointed at a patched op-node. The patch would restore the old behaviour for generating output roots.
+
+### FM2: Chain split from genesis due to future changes to genesis state tooling
+
+- **Description:**
+  The genesis state is, with the activation of Isthmus at genesis, now part of the genesis block. Therefore changes to the tooling which generates the genesis block from the genesis state can cause a chain split if there is a bug introduced in the future or even if the genesis state is changed intentionally but in an uncoordinated manner.
```

Comment:
I would love clarification on this, for me this case is related to the fact that only "old" nodes are causing a chain split with new node since some features are not included? but somehow seems to be not only related to this feature right? 
any new feature that is included would cause the same behavior. 
I am not sure to follow here, if this possible to have a more detail recap?

Reply:
I agree, I think the chain split problem is:

1. Not specific to this feature, since modifying the genesis state can cause a chain split anyway since the state root would change.
2. Not likely to come up in production since the genesis is usually created once and then never modified. @tynes can you think of a scenario (outside of testing) where the genesis is regenerated?

Reply:
Got it, clarified here [e2e8d16](https://github.com/ethereum-optimism/design-docs/pull/223/commits/e2e8d1646baf430189c8d66965a7fe9ee31024dd)

Reply:
Oh thank for clarifying here!

Reply:
As @geoknee mentioned and I completely agree we should add this one at generic FMA for HF for all the next HF in the future.

Reply:
It is done [d72d299](https://github.com/ethereum-optimism/design-docs/pull/223/commits/d72d299d1c825984652d894102fa63a4457f8d56). Please resolve if happy @Ethnical